File: bindings/csharp/MikanVideoSourceAPI.cs
using System.Threading.Tasks;

namespace MikanXR
{	
	public class MikanVideoSourceAPI
	{
		private MikanRequestManager _requestManager;

		public MikanVideoSourceAPI(MikanRequestManager requestManager)
		{
			_requestManager= requestManager;
			_requestManager.AddResponseFactory<MikanVideoSourceIntrinsics>();
			_requestManager.AddResponseFactory<MikanVideoSourceMode>();
			_requestManager.AddResponseFactory<MikanVideoSourceAttachmentInfo>();
		}

		private static readonly string k_getVideoSourceIntrinsics = "getVideoSourceIntrinsics";
		public Task<MikanResponse> GetVideoSourceIntrinsics() // returns MikanVideoSourceIntrinsics
		{
			return _requestManager.SendRequest(k_getVideoSourceIntrinsics);
		}
		
		private static readonly string k_getVideoSourceMode = "getVideoSourceMode";
		public Task<MikanResponse> GetVideoSourceMode() // returns MikanVideoSourceMode
		{
			return _requestManager.SendRequest(k_getVideoSourceMode);
		}
		
		private static readonly string k_getVideoSourceAttachment = "getVideoSourceAttachment";
		public Task<MikanResponse> GetVideoSourceAttachment() // returns MikanVideoSourceAttachmentInfo
		{
			return _requestManager.SendRequest(k_getVideoSourceAttachment);
		}
	}
}

File: src/ClientAPI/Private/MikanVideoSourceAPI.cpp
#include "MikanVideoSourceAPI.h"
#include "MikanRequestManager.h"

MikanVideoSourceAPI::MikanVideoSourceAPI(MikanRequestManager* requestManager)
	: m_requestManager(requestManager)
{
}

MikanResponseFuture MikanVideoSourceAPI::getVideoSourceIntrinsics()
{
	return m_requestManager->sendRequest(k_getVideoSourceIntrinsics);
}

MikanResponseFuture MikanVideoSourceAPI::getVideoSourceMode()
{
	return m_requestManager->sendRequest(k_getVideoSourceMode);
}

MikanResponseFuture MikanVideoSourceAPI::getVideoSourceAttachment()
{
	return m_requestManager->sendRequest(k_getVideoSourceAttachment);
}